ext/standard/type.c

```diff
@@ -465,3 +465,176 @@ PHP_FUNCTION(is_countable)
 	RETURN_BOOL(zend_is_countable(var));
 }
 /* }}} */
+
+static bool can_int_be_exact_float(zend_long lval) {
+#if SIZEOF_ZEND_LONG == 4
+	return true;
+#else
+	if (lval >= -9007199254740992LL && lval <= 9007199254740992LL) {
```

Comment:
What am I missing? Can't this entire else block just be `return (zend_long) (double) lval == lval;` ?

Reply:
Ah I guess UB... Can't we workaround that, the one liner is so much easier and prettier